syntax = "proto3";

package identity_verification;

// Define the Identity Verification Service
service IdentityVerificationService {
  // Verify an identity based on application details
  rpc VerifyIdentity (VerificationRequest) returns (VerificationResponse) {}
  
  // Stream verification results for multiple applications
  rpc StreamVerifications (stream VerificationRequest) returns (stream VerificationResponse) {}
}

// Request message containing application details
message VerificationRequest {
  string application_id = 1;
  Applicant applicant = 2;
  Document document = 3;
}

// Applicant information
message Applicant {
  string first_name = 1;
  string last_name = 2;
  string date_of_birth = 3;
  Address address = 4;
}

// Address information
message Address {
  string street = 1;
  string city = 2;
  string state = 3;
  string postal_code = 4;
  string country = 5;
}

// Document information
message Document {
  enum DocumentType {
    PASSPORT = 0;
    DRIVERS_LICENSE = 1;
    NATIONAL_ID = 2;
    OTHER = 3;
  }
  
  DocumentType type = 1;
  string document_number = 2;
  string issuing_country = 3;
  string expiry_date = 4;
}

// Response message containing verification result
message VerificationResponse {
  string application_id = 1;
  
  enum VerificationStatus {
    VERIFIED = 0;
    REJECTED = 1;
    MANUAL_REVIEW = 2;
    ERROR = 3;
  }
  
  VerificationStatus status = 2;
  string message = 3;
  float confidence_score = 4;
  string verification_id = 5;
  string timestamp = 6;
}
